assets/js/components/user-input/UserInputQuestionWrapper.js

```diff
@@ -51,12 +51,17 @@ export default function UserInputQuestionWrapper( props ) {
 		back,
 		backLabel,
 		error,
+		allowEmptyValues,
 	} = props;
 
 	const values = useSelect( ( select ) => select( CORE_USER ).getUserInputSetting( slug ) || [] );
 	const scope = useSelect( ( select ) => select( CORE_USER ).getUserInputSettingScope( slug ) );
 	const author = useSelect( ( select ) => select( CORE_USER ).getUserInputSettingAuthor( slug ) );
 
+	const nextDisabled = allowEmptyValues
+		? values.filter( ( value ) => value.trim().length > 0 ).length === 0
+		: values.length === 0 || values.some( ( value ) => value.trim().length === 0 );
+
```

Comment:
The changes in this component seem unnecessary. The tests still pass with `allowEmptyValues` set to `true` or `false`. Both branches of the condition seem like they would result in the same result.

Reply:
Tests work because we enter all three non-empty keywords. If you comment out one of the keywords, tests will fail.

The problem with the initial approach was that it disabled the Next button on the keywords screen because there was an empty string in the values (which is a possible case for that screen). This "allowEmptyValues" property fixes this problem.

Reply:
Okay, got it; thanks for clarifying 👍

Reply:
@eugene-manuilov please add a comment to explain here as we discussed, then this should be good to go 👍

Reply:
@aaemnnosttv a comment is added.